src/main/java/bio/terra/cli/service/utils/SamService.java

```diff
@@ -81,16 +82,32 @@ private void buildClientForTerraUser() {
    */
   public SystemStatus getStatus() {
     StatusApi statusApi = new StatusApi(apiClient);
-    SystemStatus status = null;
     try {
-      status =
-          HttpUtils.callWithRetries(() -> statusApi.getSystemStatus(), SamService::isRetryable);
+      return HttpUtils.callWithRetries(() -> statusApi.getSystemStatus(), SamService::isRetryable);
     } catch (ApiException | InterruptedException ex) {
-      logger.error("Error getting SAM status", ex);
+      throw new RuntimeException("Error getting SAM status.", ex);
+    } finally {
+      closeConnectionPool();
+    }
+  }
+
+  /**
+   * Call the SAM "/api/users/v1/invite/{inviteeEmail}" endpoint to invite a user and track them.
+   * This is not the same thing as registering a user.
+   *
+   * @param userEmail email to invite
+   * @return SAM object with details about the user
+   */
+  public UserStatusDetails inviteUser(String userEmail) {
```

Comment:
does this actually communicate with the user? Is there an email template or something we need to set up?

Reply:
My understanding (from @wchamber on [Slack](https://broadinstitute.slack.com/archives/C016V54EK0B/p1614371604039900) and through testing with throwaway gmail accounts) is that there is no communication, it just gives them a subject id in SAM so that we can refer to them in the system. Then when they do register, SAM will connect them with this same subject id. So the effect is that we can "share" a workspace with a user before they actually register.